### ISSUE TITLE
Policy network matches output with input?

### ISSUE THREAD

**Issue by liuzhuang13** [wallarm/nascell-automl#5]
Thanks for the nice code and blog. But from my understanding,
```python
with tf.variable_scope("policy_network", reuse=True):
                self.logprobs = self.policy_network(self.states, self.max_layers)
                print("self.logprobs", self.logprobs)

# compute policy loss and regularization loss
self.cross_entropy_loss = tf.nn.softmax_cross_entropy_with_logits(logits=self.logprobs[:, -1, :], labels=self.states)
```
this code seems to imply the "policy_network" takes self.states as input, and the cross entropy loss matches its output with self.states. So the "policy_network" is simply regressing the input?

**Comment by liuzhuang13** [comment]
Sorry, I misunderstood the tf.nn.softmax_cross_entropy_with_logits function, please kindly ignore this issue.

**Comment by johsnows** [comment]
what did you mean misunderstood,  i think this is a problem too, can you explain why this is true?
